### PR TITLE
[PART-2] CPUAggregation Refactor : BinSorter : Make sorting optional

### PR DESCRIPTION
**Pull request by 1chandu**
For #3583

NOTE: Splitting #3912 to multiple stacked PRs, will close #3912 (keeping it open for existing discussions to be resolved).

#### Change List
- In preparation to consolidate all CPU Aggregation modules to same set of utility methods : 
   -  Make sorting of bins optional (for layers that don't have calculate lower/higher percentiles, sorting is not needed).
   - Min/Max values:  Calculate both min and max values to match GPU Aggregation.

### CODE REVIEW

File: modules/aggregation-layers/src/utils/bin-sorter.js [hunk 1]
```diff
@@ -74,16 +77,6 @@ export default class BinSorter {
     return [this.sortedBins[lowerIdx].value, this.sortedBins[upperIdx].value];
```
Comment (heshan0131) [inline modules/aggregation-layers/src/utils/bin-sorter.js]: sorting is used to quickly return value range by percentile. If the bins are not sorted, here you will have to first sort them before return value range by lower and upper percentile range

Comment (1chandu) [reply modules/aggregation-layers/src/utils/bin-sorter.js]: With this refactor, I am going to use same cpu grid aggregation methods for all the layers (earlier I was using cpu aggregation in `GPUGridAggregator`), what I am doing here is, make sorting optional (by default it is true). Layers other than `CPUGridLayer` and `HexagonLayer`, don't need sorting, they don't support percentile props.

File: modules/aggregation-layers/src/utils/bin-sorter.js [hunk 2]
```diff
@@ -64,38 +69,51 @@ export default class BinSorter {
...
 
   /**
    * Get a mapping from cell/hexagon index to sorted bin
    * This is used to retrieve bin value for color calculation
-   * @return {Object} bin index to sortedBins
+   * @return {Object} bin index to aggregatedBins
    */
   getBinMap() {
-    return this.sortedBins.reduce(
+    return this.aggregatedBins.reduce(
```
Comment (Pessimistress) [inline modules/aggregation-layers/src/utils/bin-sorter.js]: Use for loop and `mapper[curr.i] = curr`

File: modules/aggregation-layers/src/utils/bin-sorter.js [hunk 3]
```diff
@@ -64,38 +69,51 @@ export default class BinSorter {
...
+    return [this.aggregatedBins[lowerIdx].value, this.aggregatedBins[upperIdx].value];
   }
 
   /**
    * Get a mapping from cell/hexagon index to sorted bin
    * This is used to retrieve bin value for color calculation
-   * @return {Object} bin index to sortedBins
+   * @return {Object} bin index to aggregatedBins
    */
   getBinMap() {
```
Comment (Pessimistress) [inline modules/aggregation-layers/src/utils/bin-sorter.js]: Is this a public method?

File: modules/aggregation-layers/src/utils/bin-sorter.js [hunk 4]
```diff
@@ -23,37 +23,42 @@
...
    * @param {Function} getValue
+   * @param {Bool} sort
    * @return {Array} array of values and index lookup
    */
-  getSortedBins(bins, getValue) {
-    return bins
-      .reduce((accu, h, i) => {
-        const value = getValue(h.points);
+  getAggregatedBins(bins, getValue, sort) {
+    const aggregatedBins = bins.reduce((accu, h, i) => {
```
Comment (Pessimistress) [inline modules/aggregation-layers/src/utils/bin-sorter.js]: - Is this a public method? Can the user update an existing instance?
- Use for loop

File: modules/aggregation-layers/src/utils/bin-sorter.js (hunk 3 above)
Comment (1chandu) [reply modules/aggregation-layers/src/utils/bin-sorter.js]: @heshan0131 is going to mark private/public methods in this file.

File: modules/aggregation-layers/src/utils/bin-sorter.js (hunk 1 above)
Comment (1chandu) [reply modules/aggregation-layers/src/utils/bin-sorter.js]: as discussed offline, added code to compute sortedBins on demand and save it to the class, removed `sort` flag.

**Pull request merged**